docker/build-tools/Dockerfile

```diff
@@ -0,0 +1,91 @@
+RUN apt-get update
+RUN apt-get install -y build-essential unzip autoconf libtool zlibc zlib1g-dev libc-ares-dev libssl-dev upx
+
+# Prepare common output directories
+RUN mkdir -p /usr/bin
+RUN mkdir -p /usr/include
+
+# Install protoc
+WORKDIR /tmp
+ADD https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOBUF_VERSION}/protoc-${PROTOBUF_VERSION}-linux-x86_64.zip /tmp/
+RUN unzip /tmp/protoc-${PROTOBUF_VERSION}-linux-x86_64.zip
+RUN cp -aR /tmp/bin/protoc /usr/bin
+
+# Install necessary protoc includes
+RUN mkdir -p /usr/include/protobuf/google/protobuf
+RUN mkdir -p /usr/include/protobuf/google/rpc
+RUN mkdir -p /usr/include/protobuf/gogoproto
+RUN for f in any duration descriptor empty struct timestamp wrappers; do \
+            curl -L -o /usr/include/protobuf/google/protobuf/${f}.proto https://raw.githubusercontent.com/google/protobuf/master/src/google/protobuf/${f}.proto; \
+        done
+RUN for f in code error_details status http; do \
+            curl -L -o /usr/include/protobuf/google/rpc/${f}.proto https://raw.githubusercontent.com/istio/gogo-genproto/master/googleapis/google/rpc/${f}.proto; \
+        done
+RUN curl -L -o /usr/include/protobuf/gogoproto/gogo.proto https://raw.githubusercontent.com/gogo/protobuf/master/gogoproto/gogo.proto
+
+# Install external dependencies
+RUN GO111MODULE=on go get github.com/golang/protobuf/protoc-gen-go@${GOLANG_PROTOBUF_VERSION}
+RUN GO111MODULE=on go get github.com/istio/gogo-protobuf/protoc-gen-gogofast@${GOGO_PROTOBUF_VERSION}
+RUN GO111MODULE=on go get github.com/istio/gogo-protobuf/protoc-gen-gogofaster@${GOGO_PROTOBUF_VERSION}
+RUN GO111MODULE=on go get github.com/istio/gogo-protobuf/protoc-gen-gogoslick@${GOGO_PROTOBUF_VERSION}
+RUN GO111MODULE=on go get github.com/uber/prototool/cmd/prototool@${PROTOTOOL_VERSION}
+RUN GO111MODULE=on go get github.com/nilslice/protolock/cmd/protolock@${PROTOLOCK_VERSION}
+RUN GO111MODULE=on go get github.com/maxbrunsfeld/counterfeiter/v6@${COUNTERFEITER_VERSION}
+RUN GO111MODULE=on go get golang.org/x/tools/cmd/goimports@${GOIMPORTS_VERSION}
+RUN GO111MODULE=on go get github.com/golangci/golangci-lint/cmd/golangci-lint@${GOLANGCI_LINT_VERSION}
+RUN go get github.com/jteeuwen/go-bindata/go-bindata
+
+# Install latest versions of custom Istio tools
+RUN go get istio.io/tools/protoc-gen-docs
+RUN go get istio.io/tools/cmd/annotations_prep
+
+# Save some bytes in the container
+# RUN upx --lzma /go/bin/*
+
+# Prewarm the golang module cache
+RUN rm -fr /go/pkg/mod/*
+RUN mkdir -p /home
+WORKDIR /home
+COPY go.mod /home
+RUN go mod download
+
+# Cleanup
+WORKDIR /
+RUN apt-get purge -y unzip autoconf libtool zlibc zlib1g-dev libc-ares-dev libssl-dev upx
+RUN apt-get autoremove -y
+RUN apt-get clean
+RUN rm -fr /root/.cache/go-build
+RUN rm -fr /home
+RUN rm -fr /tmp/*
+
+# TODO: sadly this doesn't work as some processes launched in the container won't start. For example, make fails with the super useful:
+#
+#   make: *** [gen] Error 139
```

Comment:
How do I do this?

Reply:
run `ldconfig` - this will write out the new ld.so cache based upon shared libs in the system.  I take a look at your pr in my system and see if I can get things working.